tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -53,6 +53,83 @@ static void check_out_type_matches(Tensor result,
   }
 }
 
+inline Tensor & dispatch_arange(Scalar end, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, end);
+}
+
+inline Tensor dispatch_arange(Scalar end, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, end);
+}
+
+inline Tensor & dispatch_arange(Scalar start, Scalar end, Scalar step, Tensor result) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(result);
+  return at::arange_out(result, start, end, step);
+}
+
+inline Tensor dispatch_arange(Scalar start, Scalar end, Scalar step, const Type & dtype, int64_t device) {
+  maybe_initialize_cuda(dtype);
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(device);
+  return at::arange(dtype, start, end, step);
+}
+
+static inline bool allIntegral(std::initializer_list<Scalar> l) {
+  for (auto s : l) {
+    if (!s.isIntegral() && !(s.isBackedByTensor() && at::isIntegralType(s.toTensor().type().scalarType()))) {
+      return false;
+    }
+  }
+  return true;
+}
+
+static PyObject * THPVariable_arange(PyObject* self, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "arange(Scalar end, *, Tensor out=None, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool requires_grad=False)",
+    "arange(Scalar start, Scalar end, Scalar step=1, *, Tensor out=None, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool requires_grad=False)",
+  });
+
+  ParsedArgs<8> parsed_args;
+  auto r = parser.parse(args, kwargs, parsed_args);
+
+  if (r.idx == 0) {
+    if (r.isNone(1)) {
+      auto device = r.device(4);
+      auto end = r.scalar(0);
+      auto scalarType = r.isNone(2) && allIntegral({end}) ? at::ScalarType::Long : r.scalartype(2);
```

Comment:
Can you do `scalartype(2)` if `isNone(2)` is true? It's possible to trigger this if `end` is not integral.